pyiron/atomistics/master/elastic.py

```diff
@@ -0,0 +1,223 @@
+__email__ = "janssen@mpie.de"
+__status__ = "production"
+__date__ = "Sep 1, 2017"
+
+
+eV_div_A3_to_GPa = (
+    1e21 / scipy.constants.physical_constants["joule-electron volt relationship"][0]
+)
+
+def calc_elastic_tensor(strain, stress=None, energy=None, volume=None, rotations=None):
+    """
+    Calculate 6x6-elastic tensor from the strain and stress or strain and energy+volume.
+
+    Rotations matrices can be added to take box symmetries into account (unit matrix can
+    be added but does not have to be included in the list)
+
+    Args:
+        strain (numpy.ndarray): nx3x3 strain tensors
+        stress (numpy.ndarray): nx3x3 stress tensors
+        energy (numpy.ndarray): n energy values
+        volume (numpy.ndarray): n volume values
+        rotations (numpy.ndarray): mx3x3 rotation matrices
+    """
+    if len(strain)==0:
+        raise ValueError('Not enough points')
+    if rotations is not None:
+        rotations = np.append(np.eye(3), rotations).reshape(-1, 3, 3)
+        _, indices = np.unique(np.round(rotations, 6), axis=0, return_inverse=True)
+        rotations = rotations[np.unique(indices)]
+    else:
+        rotations = np.eye(3)
+    strain = np.einsum('nik,mkl,njl->nmij', rotations, strain, rotations).reshape(-1, 3, 3)
+    strain = np.stack((strain[:,0,0], strain[:,1,1], strain[:,2,2], 2*strain[:,1,2], 2*strain[:,0,2], 2*strain[:,0,1]), axis=-1)
+    coeff = []
+    if stress is not None and len(stress)*len(rotations)==len(strain):
+        stress = np.einsum('nik,mkl,njl->nmij', rotations, stress, rotations).reshape(-1, 3, 3)
+        stress = np.stack((stress[:,0,0], stress[:,1,1], stress[:,2,2], stress[:,1,2], stress[:,0,2], stress[:,0,1]), axis=-1)
+        for ii in range(6):
+            reg = LinearRegression().fit(strain, stress[:,ii])
+            coeff.append(reg.coef_)
+        coeff = np.array(coeff)
+    elif energy is not None and volume is not None and len(energy)*len(rotations)==len(strain) and len(energy)==len(volume):
+        energy = np.tile(energy, len(rotations))
+        volume = np.tile(volume, len(rotations))
+        C = np.einsum('n,ni,nj->nij', 0.5*volume, strain, strain)
+        C = np.triu(C).reshape(-1, 36)
+        C = C[:,np.sum(C, axis=0)!=0]
+        C = np.append(np.ones(len(C)).reshape(-1, 1), C, axis=-1)
+        reg = LinearRegression().fit(C, energy)
+        coeff = np.triu(np.ones((6,6))).flatten()
+        coeff[coeff!=0] *= reg.coef_[1:]*eV_div_A3_to_GPa
+        coeff = coeff.reshape(6,6)
+        coeff = 0.5*(coeff+coeff.T)
+    else:
+        raise ValueError('Problem with fitting data')
+    return coeff
+
+def calc_elastic_constants(elastic_tensor):
+    """
+    Calculate elastic constants from the elastic tensor.
+
+    For anistropic material (i.e. zener_ratio!=1), the values may or may not make
+    sense -> don't trust the results straightforwardly
+
+    Args:
+        elastic_tensor (numpy.ndarray): 6x6 tensor
+    """
+    output = {}
```

Comment:
Should this be an `InputList` instead of a dict? I'm not sure what the current big picture is, but we should start looking for these opportunities.

Reply:
Also, if this is only a dict does it end up in the HDF (not that an InputList would help in this regard yet)?

Reply:
No it doesn't, it's just a function which returns output values. I guess it has a solid utility since you might be having an elastic tensor from somewhere and you might want to have elastic constants. For this, it makes more sense to have an independent function.